workshops/PREFACE.md

```diff
@@ -0,0 +1,27 @@
+# Preface
+
+_Adapted from [here](https://medium.freecodecamp.com/things-i-wish-someone-had-told-me-when-i-was-learning-how-to-code-565fc9dcb329)_
+
+---
+
+If you're reading this, that means that you've decided to embark on the journey of learning to code. Congratulations and welcome, we're excited to have you.
+
+## Coding is Not Special
+
+First and foremost, coding is a skill like any other. Like learning a language, there's grammar and vocabulary to acquire. Like math, there are processes to work through specific types of problems. Like all kinds of craftsmanship and art-making, there are techniques and tools and best practices that people have developed over time, specialized to different tasks, that you're free to use or modify or discard.
+
+Someone will always tell you you're doing it wrong. There are almost always many different approaches to a particular problem with no single "right way". A lot of programmers get very good at advocating for their preferred way, but that doesn't mean it's the One Truth Path.
+
+## It Never Works the First Time
+
+When you first start learning to code, you'll very quickly run up against this particular experience: you think you've set up everything the way you're supposed to, you've checked and re-checked it, and it still. doesn't. work. You don't have a clue where to begin trying to fix it, and the error message (if you're lucky enough to have one at all) might as well say "fuck you". You might be tempted to give up at this point, thinking that you'll never figure it out, that you're not cut out for this.
+
+But this experience is so common for programmers of all skill levels that it says absolutely nothing about your intelligence, tech-savviness, or suitability for the coding life. It will happen to you as a beginner, but it will also happen to you as an experienced programmer. The main difference will be in how you respond to it.
+
+A big difference between new coders and experienced coders is faith: faith that things are going wrong for a logical and discoverable reason, faith that problems are fixable, faith that there is a way to accomplish the goal. The path from "not working" to "working" might not be obvious, but with patience you can usually find it.
+
+## Sticking with it is More Important Than the Method
```

Comment:
Gotta update my emacs function for this.

On Mon, Apr 18, 2016 at 10:18 PM, JESSICA <notifications@github.com> wrote:

> In workshops/PREFACE.md
> <https://github.com/hackclub/hackclub/pull/790#discussion_r60174198>:
>
> > +
> > +## Coding is Not Special
> > +
> > +First and foremost, coding is a skill like any other. Like learning a language, there's grammar and vocabulary to acquire. Like math, there are processes to work through specific types of problems. Like all kinds of craftsmanship and art-making, there are techniques and tools and best practices that people have developed over time, specialized to different tasks, that you're free to use or modify or discard.
> > +
> > +Someone will always tell you you're doing it wrong. There are almost always many different approaches to a particular problem with no single "right way". A lot of programmers get very good at advocating for their preferred way, but that doesn't mean it's the One Truth Path.
> > +
> > +## It Never Works the First Time
> > +
> > +When you first start learning to code, you'll very quickly run up against this particular experience: you think you've set up everything the way you're supposed to, you've checked and re-checked it, and it still. doesn't. work. You don't have a clue where to begin trying to fix it, and the error message (if you're lucky enough to have one at all) might as well say "fuck you". You might be tempted to give up at this point, thinking that you'll never figure it out, that you're not cut out for this.
> > +
> > +But this experience is so common for programmers of all skill levels that it says absolutely nothing about your intelligence, tech-savviness, or suitability for the coding life. It will happen to you as a beginner, but it will also happen to you as an experienced programmer. The main difference will be in how you respond to it.
> > +
> > +A big difference between new coders and experienced coders is faith: faith that things are going wrong for a logical and discoverable reason, faith that problems are fixable, faith that there is a way to accomplish the goal. The path from "not working" to "working" might not be obvious, but with patience you can usually find it.
> > +
> > +## Sticking with it is More Important Than the Method
>
> wow titlecase failures ova here
> *Sticking with It Is More Important than the Method*
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/hackclub/hackclub/pull/790/files/b7f32a7938869d17222be04b32487f07d4935bfe#r60174198>
>



-- 
Zach Latta
https://zachlatta.com <http://zachlatta.com>